samples/appendable-template/README.md

```diff
@@ -0,0 +1,194 @@
+
+```python
+    train_step = TrainStep(workspace=aml_workspace, env=e,
+                           compute=aml_compute, config=run_config,
+                           pipeline_parameters=pipeline_parameters,
+                           output_pipelinedata=pipeline_data)
+    train_step.append_step(steps)
+
+    register_step = RegisterStep(workspace=aml_workspace, env=e,
+                                 compute=aml_compute, config=run_config,
+                                 pipeline_parameters=pipeline_parameters,
+                                 input_pipelinedata=pipeline_data)
+    register_step.append_step(steps)
+
+    train_pipeline = Pipeline(workspace=aml_workspace, steps=steps)
+    train_pipeline._set_experiment_name
+    train_pipeline.validate()
+    published_pipeline = train_pipeline.publish(
+        name=e.pipeline_name,
+        description="Model training/retraining pipeline",
+        version=e.build_id,
+    )
+```
+
+This is an example of a class for a pipeline step. This class encapsulates details of a pipeline step. When multiple developers work on multiple pipeline steps, each developer can work on a specific class for a pipeline step they are working on independently, instead of working on a single script (which leads to merge conflicts).
+
+```python
+from azureml.pipeline.steps import PythonScriptStep
+
+
+class TrainStep:
+    def __init__(self, workspace, env, compute, config, pipeline_parameters,
+                 output_pipelinedata):
+        self.workspace = workspace
+        self.env = env
+        self.compute = compute
+        self.config = config
+        self.pipeline_parameters = pipeline_parameters
+        self.output_pipelinedata = output_pipelinedata
+
+    def append_step(self, step_list):
+        train_step = PythonScriptStep(
+            name="Train Model",
+            compute_target=self.compute,
+            source_directory=self.env.sources_directory_train,
+            script_name=self.env.train_script_path,
+            outputs=[self.output_pipelinedata],
+            arguments=[
+                "--model_name",
+                self.pipeline_parameters["model_name"],
+                "--step_output",
+                self.output_pipelinedata,
+                "--dataset_version",
+                self.pipeline_parameters["dataset_version"],
+                "--data_file_path",
+                self.pipeline_parameters["data_file_path"],
+                "--caller_run_id",
+                self.pipeline_parameters["caller_run_id"],
+                "--dataset_name",
+                self.env.dataset_name,
+            ],
+            runconfig=self.config,
+            allow_reuse=True,
+        )
+
+        if len(step_list) > 0:
+            previous_step = step_list[-1]
+            train_step.run_after(previous_step)
+
+        step_list.append(train_step)
+```
+
+# Getting Started
+
+## Modification
```

Comment:
I would recommend not to keep both versions in the code. We can link/refer to MLOpsPython for the original non appendable template version.